Core/Core/Router/Router.swift

```diff
@@ -181,6 +181,12 @@ public class Router: RouterProtocol {
         DeveloperMenuViewController.recordRouteInHistory(url.url?.absoluteString)
         #endif
         Analytics.shared.logEvent("route", parameters: ["url": String(describing: url)])
+
+        if url.host?.isEmpty == false && !urlMatchesSessionHost(url) {
```

Comment:
awesome. This way we never route to an external url that happens to match one of our paths.